api.js

```diff
@@ -202,17 +216,42 @@ exports.setApp = function (app, client)
 
     app.post('/api/getGroupInfo', async (req, res, next) =>
     {
-        // incoming: groupId
-        // outgoing: event, eventName, eventPriceMin, eventPriceMax, eventDate, secretShopper_buyers, secretShopper_receivers, members[firstName, lastName, userId], error, success
+        // incoming: groupId, userId
+        // outgoing: event, eventName, eventPriceMin, eventPriceMax, eventDate, secretShopper_receiver, members[firstName, lastName, userId], error, success
 
-        const { groupId } = req.body;
+        const { groupId, userId } = req.body;
         var gid = require('mongodb').ObjectID(groupId);
 
         const db = client.db();
         try
         {
             const results = await db.collection('Groups').find({ _id:gid}).toArray();
 
+            var index = -1;
+            for(i=0; i <results[0].members.length; i++)
```

Comment:
Nothing here is wrong, but just a suggestion that can sometimes make code faster: 

When you're accessing a list, in this case "results", several times like results[0] or even results[0].members. It can help to save it in another variable so you don't keep having to keep indexing the list every time. Similarly with the length function, it's easier to calculate the length before the loop, so we don't have to keep calculating the same value. For example here, you could do the following:

```
var members = results[0].members;
var len = members.length;

for (i = 0; i < len ; i++) {
  if (members[i] == userId)
    index = i;
}
```